src/metabase/driver/sql/parameters/substitution.clj

```diff
@@ -89,19 +89,34 @@
   [_driver t]
   (make-stmt-subs "?" [t]))
 
+(defmulti align-temporal-unit-with-param-type-and-value
+  "Returns a suitable temporal unit conversion keyword for `field`, `param-type`, `value` and the given driver.
+  The resulting keyword will be used to call the corresponding `metabase.driver.sql.query-processor/date`
+  implementation to convert the `field`.
+  Returns `nil` if the conversion is not necessary for this `field`, `param-type` and `value` combination."
+  {:added "0.49.0" :arglists '([driver field param-type value])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
+
 (defmulti align-temporal-unit-with-param-type
-  "Returns a suitable temporal unit conversion keyword for `field`, `param-type` and the given driver. The resulting keyword
-  will be used to call the corresponding `metabase.driver.sql.query-processor/date` implementation to convert the `field`.
-  Returns `nil` if the conversion is not necessary for this `field` and `param-type` combination."
-  {:added "0.48.0" :arglists '([driver field param-type])}
+  "Returns a suitable temporal unit conversion keyword for `field`, `param-type` and the given driver.
+  The resulting keyword will be used to call the corresponding `metabase.driver.sql.query-processor/date`
+  implementation to convert the `field`.
+  Returns `nil` if the conversion is not necessary for this `field` and `param-type` combination.
+  Deprecated: use `align-temporal-unit-with-param-type-and-value` instead, as it has access to `value`."
+  {:added "0.48.0" :deprecated "0.49.0" :arglists '([driver field param-type])}
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
-(defmethod align-temporal-unit-with-param-type :default
-  [_driver _field param-type]
+(defmethod align-temporal-unit-with-param-type-and-value :default
+  [_driver _field param-type _value]
   (when (params.dates/date-type? param-type)
     :day))
 
+#_{:clj-kondo/ignore [:deprecated-var]}
+(defmethod align-temporal-unit-with-param-type :default
+  [driver _field param-type]
+  (align-temporal-unit-with-param-type-and-value driver param-type nil))
```

Comment:
These seem to be wrong way around. If a driver implemented `align-temporal-unit-with-param-type` but not yet `align-temporal-unit-with-param-type-and-value`, then it wouldn't get called. For the time being, the default `align-temporal-unit-with-param-type-and-value` has to forward to `align-temporal-unit-with-param-type`.

Reply:
Fixed, thanks! :+1: